Spring cleaning

kierdavis opened the pull request:
Deprecates the "TFoo" naming convention in favour of simply "Foo".

Also updates instances of "Nimrod" to "Nim".

kierdavis committed:
Remove the 'T' prefix from all types that have it

See also: https://github.com/nim-lang/Nim/issues/2773

Additionally, names of types that previously used a mixture of "IRC" and "Irc" have been fixed to consistently use only "IRC".

kierdavis committed:
s/Nimrod/Nim/

dom96 commented:
This is awesome, thanks!

But please restore the acronym convention, i.e. ``IrcBase`` instead of ``IRCBase``. Also, could you fix every usage of the prefixed type in the .nim files as well?

kierdavis committed:
Add a .gitignore

kierdavis committed:
Use 'Irc' naming convention instead of 'IRC'

kierdavis commented:
My bad, wasn't sure which convention ("Irc" or "IRC") was the preferred one.

I'm not sure what you mean by your second point - are you referring to changing all the types prefixed with "P" as well?

dom96 commented:
> I'm not sure what you mean by your second point - are you referring to changing all the types prefixed with "P" as well?

Yes.

kierdavis committed:
Remove the 'P' prefix from all types that have it

This involved renaming 'IrcBase' to 'IrcBaseObj', to avoid a clash when renaming 'PIrcBase' to 'IrcBase'.

kierdavis committed:
Fix a compiler warning resulting from a non-ordinal case block not having an else branch

kierdavis committed:
Update irc.nimble to the new NimScript format

kierdavis commented:
I've changed the P types too now. There was a clash, so the type originally prefixed with 'T' ('TIrcBase') was renamed to 'IrcBaseObj'.

I've also fixed a compiler warning (non-ordinal case missing an else branch) and updated the Nimble file to the NimScript format.

dom96 commented:
Thanks!
